paldb/src/main/java/com/linkedin/paldb/impl/StorageReader.java

```diff
@@ -83,9 +85,6 @@
   private FileChannel channel;
   // Use MMap for data?
   private final boolean mMapData;
-  // Buffers
-  private final DataInputOutput sizeBuffer = new DataInputOutput(new byte[5]);
-  private final byte[] slotBuffer;
```

Comment:
Using a ThreadLocal<byte[]> (perhaps static, with max size being used for all slot sizes) would improve performance, reducing allocation.   The same goes for the `DataInputOutput`s.   Perhaps a single class, containing all of the necessary buffers that a thread needs (arrays, DataInputOutputs, etc) and a single ThreadLocal holding the per thread buffers would perform best.